src/sentry/api/serializers/models/incidentactivity.py

```diff
@@ -6,32 +6,50 @@
     Serializer,
     register,
 )
-from sentry.incidents.models import (
-    Incident,
-    IncidentActivity,
-)
+from sentry.utils.snuba import SnubaTSResult
+from sentry.api.serializers import serialize
+from sentry.api.serializers.snuba import SnubaTSResultSerializer
+from sentry.incidents.models import IncidentActivity
+from sentry.utils.db import attach_foreignkey
 
 
 @register(IncidentActivity)
 class IncidentActivitySerializer(Serializer):
-    def get_attrs(self, item_list, **kwargs):
-        incidents = Incident.objects.filter(id__in=set(i.incident_id for i in item_list))
-        incident_lookup = {incident.id: incident for incident in incidents}
+    def get_attrs(self, item_list, user, **kwargs):
+        attach_foreignkey(item_list, IncidentActivity.incident, related=('organization',))
+        attach_foreignkey(item_list, IncidentActivity.event_stats_snapshot)
+        attach_foreignkey(item_list, IncidentActivity.user)
+        serialized_users = serialize(
+            set(item.user for item in item_list if item.user_id),
+            user=user,
+        )
+        user_lookup = {user['id']: user for user in serialized_users}
+        return {item: {'user': user_lookup.get(six.text_type(item.user_id))} for item in item_list}
 
-        results = {}
-        for activity in item_list:
-            results[activity] = {'incident': incident_lookup[activity.incident_id]}
+    def serialize(self, obj, attrs, user):
+        incident = obj.incident
 
-        return results
+        event_stats = None
+        if obj.event_stats_snapshot:
+            serializer = SnubaTSResultSerializer(
+                obj.incident.organization,
+                None,
+                user,
+            )
+            event_stats = serializer.serialize(SnubaTSResult(
+                obj.event_stats_snapshot.snuba_values,
+                obj.event_stats_snapshot.start,
+                obj.event_stats_snapshot.end,
+                obj.event_stats_snapshot.period,
+            ))
 
-    def serialize(self, obj, attrs, user):
-        incident = attrs['incident']
         return {
             'id': six.text_type(obj.id),
             'incidentIdentifier': six.text_type(incident.identifier),
-            'userId': six.text_type(obj.user_id),
+            'user': attrs['user'],
             'type': obj.type,
             'value': obj.value,
             'previousValue': obj.previous_value,
             'comment': obj.comment,
+            'eventStats': event_stats,
```

Comment:
We're also missing a date (we use `dateCreated` with Issue activity items):

```suggestion
            'eventStats': event_stats,
            'dateCreated': obj.date_added,
```